File: TheepotAndroid/app/src/main/kotlin/nl/bsodetheepot/mobile/data/models/Profiel.kt
package nl.bsodetheepot.mobile.data.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/** Spiegelt de `profielen`-tabel en het `Rol`-type uit lib/supabase.ts. */
@Serializable
enum class Rol {
    @SerialName("superadmin") SUPERADMIN,
    @SerialName("directie") DIRECTIE,
    @SerialName("leidinggevende") LEIDINGGEVENDE,
    @SerialName("locatie") LOCATIE;

    val label: String
        get() = when (this) {
            SUPERADMIN -> "Superadmin"
            DIRECTIE -> "Directie"
            LEIDINGGEVENDE -> "Leidinggevende"
            LOCATIE -> "Locatie"
        }
}

@Serializable
data class Profiel(
    val id: String,
    val email: String,
    val naam: String,
    val rol: Rol,
    val actief: Boolean,
    @SerialName("aangemaakt_op") val aangemaaktOp: String,
)

File: TheepotMobile/TheepotMobile/Models/Profiel.swift
import Foundation

/// Spiegelt de `profielen`-tabel en `Rol`-type uit lib/supabase.ts.
enum Rol: String, Codable, Equatable {
    case superadmin
    case directie
    case leidinggevende
    case locatie

    var label: String {
        switch self {
        case .superadmin: return "Superadmin"
        case .directie: return "Directie"
        case .leidinggevende: return "Leidinggevende"
        case .locatie: return "Locatie"
        }
    }
}

struct Profiel: Codable, Identifiable {
    let id: UUID
    let email: String
    let naam: String
    let rol: Rol
    let actief: Bool
    let aangemaaktOp: String

    enum CodingKeys: String, CodingKey {
        case id, email, naam, rol, actief
        case aangemaaktOp = "aangemaakt_op"
    }
}
